### PR TITLE
Update protobuf plugin

### PR DESCRIPTION
**Pull request by loosebazooka**
Looks like the new protobuf plugin handles a few more things that the old one. Updates to remove unnecessary config that was breaking things.

Signed-off-by: Appu Goundan <appu@google.com>

### CODE REVIEW

**Review by patflynn — APPROVED** [review]

**Comment by loosebazooka** [comment]
I think staging is going through something, I'll wait for it come back and run these tests again.

File: sigstore-java/build.gradle.kts [hunk 1]
```diff
@@ -67,7 +63,7 @@ protobuf {
     generateProtoTasks {
         ofSourceSet("main").forEach() {
             it.plugins {
-                id("grpc")
+                id("grpc") {}
             }
         }
```
Comment (vlsi) [inline sigstore-java/build.gradle.kts:68]: Please use the following here:

```kotlin
    generateProtoTasks {
        ofSourceSet("main").configureEach {
            plugins {
                id("grpc")
            }
        }
    }
```

**Comment by vlsi** [comment]
@loosebazooka , thanks.
Please replace `dependsOn("generateProto", "generateJsonSchema2DataClass0")` with `dependsOn("generateJsonSchema2DataClass0")` at the end of `/sigstore-java/build.gradle.kts`.
It looks like `dependsOn("generateProto")` is no longer needed.

File: sigstore-java/build.gradle.kts (hunk 1 above)
Comment (loosebazooka) [reply sigstore-java/build.gradle.kts:68]: Oops must've left that in there while experimenting

Comment (vlsi) [reply sigstore-java/build.gradle.kts]: I would suggest replacing `forEach()` with `configureEach`, so `it.` is not needed before `plugins`

Comment (loosebazooka) [reply sigstore-java/build.gradle.kts]: Damn, I missed that whole suggestion

Comment (vlsi) [reply sigstore-java/build.gradle.kts]: I logged an issue for streamlining protobuf-gradle-plugin configuration: https://github.com/google/protobuf-gradle-plugin/issues/628

**Review by vlsi — APPROVED** [review]

**Comment by loosebazooka** [comment]
@vlsi thanks for filing those protobuf gradle plugin bugs, I was also gonna see if they would update their kotlin-dsl example.

**Pull request merged**